src/beatstar/database.rs

```diff
@@ -143,6 +143,65 @@ pub fn beatstar_zip_content(response: ureq::Response) -> anyhow::Result<Vec<Beat
     Ok(json)
 }
 
+pub fn beatstar_zip_content_from_bytes(bytes: Vec<u8>) -> anyhow::Result<Vec<BeatStarSong>> {
```

Comment:
Why make this an entirely new method with duplication when you can migrate the common functionality into a single method?